src/components/Entities/EntityDetails.tsx

```diff
@@ -0,0 +1,97 @@
+import { Dialog } from '@material-ui/core';
+import { makeStyles, Theme } from '@material-ui/core/styles';
+import { contentMarginGridUnits } from 'common/layout';
+import { WaitForData } from 'components/common';
+import { EntityDescription } from 'components/Entities/EntityDescription';
+import { useProject } from 'components/hooks';
+import { LaunchWorkflowForm } from 'components/Launch/LaunchWorkflowForm/LaunchWorkflowForm';
+import { ResourceIdentifier } from 'models';
+import * as React from 'react';
+import { entitySections } from './constants';
+import { EntityDetailsHeader } from './EntityDetailsHeader';
+import { EntityExecutions } from './EntityExecutions';
+import { EntitySchedules } from './EntitySchedules';
+
+const useStyles = makeStyles((theme: Theme) => ({
+    metadataContainer: {
+        display: 'flex',
+        marginBottom: theme.spacing(5),
+        marginTop: theme.spacing(2),
+        width: '100%'
+    },
+    descriptionContainer: {
+        flex: '2 1 auto',
+        marginRight: theme.spacing(2)
+    },
+    executionsContainer: {
+        display: 'flex',
+        flex: '1 1 auto',
+        flexDirection: 'column',
+        margin: `0 -${theme.spacing(contentMarginGridUnits)}px`
+    },
+    schedulesContainer: {
+        flex: '1 2 auto',
+        marginRight: theme.spacing(30)
+    }
+}));
+
+export interface EntityDetailsProps {
+    id: ResourceIdentifier;
+}
+
+/** A view which optionally renders description, schedules, executions, and a
+ * launch button/form for a given entity. Note: not all components are suitable
+ * for use with all entities (not all entities have schedules, for example).
+ */
+export const EntityDetails: React.FC<EntityDetailsProps> = ({ id }) => {
+    const sections = entitySections[id.resourceType];
+    const project = useProject(id.project);
+    const styles = useStyles();
+    const [showLaunchForm, setShowLaunchForm] = React.useState(false);
+    const onLaunch = () => setShowLaunchForm(true);
+    const onCancelLaunch = () => setShowLaunchForm(false);
+
+    return (
+        <>
```

Comment:
Nit: is the fragment necessary?

Reply:
Oh hm, good catch. It's probably not necessary. My only thought is that it might have to do with some obnoxious flexbox positioning stuff in the container above this. I didn't look closely at it when porting this component over.
I'll remove it in the next PR if it turns out not to be needed. If it _is_ needed, I'll be sure to document it or see if I can come up with a better solution.